cmd/soroban-cli/src/signer.rs

```diff
@@ -189,29 +192,72 @@ fn sign_soroban_authorization_entry(
     Ok(auth)
 }
 
-pub fn sign_tx(
-    key: &ed25519_dalek::SigningKey,
-    tx: &Transaction,
-    network_passphrase: &str,
-) -> Result<TransactionEnvelope, Error> {
-    let tx_hash = hash(tx, network_passphrase)?;
-    let tx_signature = key.sign(&tx_hash);
+pub struct Signer {
+    pub kind: SignerKind,
+    pub printer: Print,
+}
 
-    let decorated_signature = DecoratedSignature {
-        hint: SignatureHint(key.verifying_key().to_bytes()[28..].try_into()?),
-        signature: Signature(tx_signature.to_bytes().try_into()?),
-    };
+#[allow(clippy::module_name_repetitions)]
+pub enum SignerKind {
+    Local(LocalKey),
+}
+
+impl Signer {
+    pub fn sign_tx(
+        &self,
+        tx: Transaction,
+        network: &Network,
+    ) -> Result<TransactionEnvelope, Error> {
+        let tx_env = TransactionEnvelope::Tx(TransactionV1Envelope {
+            tx,
+            signatures: VecM::default(),
+        });
+        self.sign_tx_env(tx_env, network)
+    }
 
-    Ok(TransactionEnvelope::Tx(TransactionV1Envelope {
-        tx: tx.clone(),
-        signatures: [decorated_signature].try_into()?,
-    }))
+    pub fn sign_tx_env(
+        &self,
+        tx_env: TransactionEnvelope,
+        network: &Network,
+    ) -> Result<TransactionEnvelope, Error> {
+        match tx_env {
+            TransactionEnvelope::Tx(TransactionV1Envelope { tx, signatures }) => {
+                let tx_hash = transaction_hash(&tx, &network.network_passphrase)?;
+                self.printer.infoln(format!(
+                    "Signing transaction with hash: {}",
+                    hex::encode(tx_hash)
+                ));
+                let decorated_signature = match &self.kind {
+                    SignerKind::Local(key) => key.sign_tx_hash(tx_hash)?,
+                };
+                let mut sigs = signatures.into_vec();
+                sigs.push(decorated_signature);
+                Ok(TransactionEnvelope::Tx(TransactionV1Envelope {
+                    tx,
+                    signatures: sigs.try_into()?,
+                }))
+            }
+            _ => Err(Error::UnsupportedTransactionEnvelopeType),
+        }
+    }
 }
 
-pub fn hash(tx: &Transaction, network_passphrase: &str) -> Result<[u8; 32], xdr::Error> {
-    let signature_payload = TransactionSignaturePayload {
-        network_id: Hash(Sha256::digest(network_passphrase).into()),
-        tagged_transaction: TransactionSignaturePayloadTaggedTransaction::Tx(tx.clone()),
-    };
-    Ok(Sha256::digest(signature_payload.to_xdr(Limits::none())?).into())
+pub struct LocalKey {
+    key: ed25519_dalek::SigningKey,
+    #[allow(dead_code)]
+    prompt: bool,
```

Comment:
Do we not intend to use prompt property here for now?

Reply:
It'll be used in a future change. We could remove it now though. I'll remove itand we can bring it back.